scripts/check_material_profiles_new_with_lxml.py

```diff
@@ -51,14 +54,15 @@ def _getAllMaterialsContentsInDir(self, directory: str) -> Dict[str, Optional[st
             break
         return result
 
-    def validateAll(self, directory: str) -> bool:
-        materials_dir = self._getMaterialsDir(os.path.abspath(directory))
+    def validateAll(self, repo_root_dir: str) -> bool:
+        materials_dir = self._getMaterialsDir(repo_root_dir)
+        fdmmaterial_xsd_file_path = os.path.join(repo_root_dir, "scripts", "fdmmaterial.xsd")
```

Comment:
You should still call `os.path.abspath` on the result here, lest the `repo_root_dir` is a relative path.

Reply:
Fixed in 3796512